src/solver/conv_asm_implicit_gemm_gtc_bwd.cpp

```diff
@@ -0,0 +1,323 @@
+        { "bwd", "fp32",   1,   0,  16, 128,   8,  16,  64,   1,   1,   1,   1,   {1,   1,   1,   1},   {1,   8,   1,  16},   {1,   1,   8,   1},   {1,   8,   1,  16},   0},
+        { "bwd", "fp32",   1,   1,  16,  64,  16,   4,  64,   1,   1,   1,   1,   {1,   1,   1,   1},   {1,  16,   1,  16},   {1,   1,   4,   1},   {1,  16,   1,  16},   0},
+        { "bwd", "fp32",   1,   0,  16,  64,  16,   4,  64,   1,   1,   1,   1,   {1,   1,   1,   1},   {1,  16,   1,  16},   {1,   1,   4,   1},   {1,  16,   1,  16},   0},
+        { "bwd", "fp32",   1,   1,  16,  32,  16,   8,  32,   1,   1,   1,   1,   {1,   1,   2,   1},   {1,  16,   1,   8},   {1,   1,   4,   1},   {1,  16,   1,   8},   0},
+        { "bwd", "fp32",   1,   0,  16,  32,  16,   8,  32,   1,   1,   1,   1,   {1,   1,   2,   1},   {1,  16,   1,   8},   {1,   1,   4,   1},   {1,  16,   1,   8},   0},
+        { "bwd", "fp32",   1,   1,  16,  32,   4,   8,  32,   2,   1,   1,   1,   {1,   1,   1,   1},   {1,   4,   1,  16},   {1,   1,   2,   1},   {1,   4,   1,  16},   0},
+        { "bwd", "fp32",   4,   1,  16,  32,   4,   8,  32,   2,   1,   1,   1,   {1,   1,   1,   1},   {1,   4,   1,  16},   {1,   1,   2,   1},   {1,   4,   1,  16},   0},
+        { "bwd", "fp32",   8,   1,  16,  32,   4,   8,  32,   2,   1,   1,   1,   {1,   1,   1,   1},   {1,   4,   1,  16},   {1,   1,   2,   1},   {1,   4,   1,  16},   0},
+        { "bwd", "fp32",  16,   1,  16,  32,   4,   8,  32,   2,   1,   1,   1,   {1,   1,   1,   1},   {1,   4,   1,  16},   {1,   1,   2,   1},   {1,   4,   1,  16},   0},
+        { "bwd", "fp32",   1,   1,   8,  64,  16,   4,  64,   1,   1,   1,   1,   {1,   1,   1,   1},   {1,  16,   1,   8},   {1,   1,   8,   1},   {1,  16,   1,   8},   0},
+        { "bwd", "fp32",   1,   0,   8,  64,  16,   4,  64,   1,   1,   1,   1,   {1,   1,   1,   1},   {1,  16,   1,   8},   {1,   1,   8,   1},   {1,  16,   1,   8},   0},
+        { "bwd", "fp32",   1,   1,   4,  64,  16,   4,  64,   1,   1,   1,   1,   {1,   1,   1,   1},   {1,  16,   1,   4},   {1,   1,  16,   1},   {1,  16,   1,   4},   0},
+        { "bwd", "fp32",   1,   0,   4,  64,  16,   4,  64,   1,   1,   1,   1,   {1,   1,   1,   1},   {1,  16,   1,   4},   {1,   1,  16,   1},   {1,  16,   1,   4},   0},
+    };
+    // clang-format on
+    return tunables;
+}
+
+/*
+* Return with true if kernel is found, and its kernel_name, block_size, grid_size as well.
+* Return with false if no kernel can be executed with input ConvolutionContext
+*/
+static inline bool FindImplicitGemmGtcDynamicBwdKernel(const ConvolutionContext& ctx,
+                                                       std::string& kernel_name,
+                                                       int& block_size,
+                                                       int& grid_size)
```

Comment:
Here maybe use a std::tuple as return type which contains valid, kernel_name, block_size and grid_size.